test/plugins/test-trace-connect.js

```diff
@@ -190,6 +190,35 @@ describe('test-trace-connect', function() {
       });
     });
   });
+
+  it('should end spans even if client aborts', function(done) {
+    var app = connect();
+    app.use('/', function (req, res) {
+      setTimeout(function() {
+        res.end(common.serverRes);
+        setImmediate(function() {
+          var traces = common.getTraces(agent);
+          assert.strictEqual(traces.length, 1);
+          assert.strictEqual(traces[0].spans.length, 1);
+          var span = traces[0].spans[0];
+          common.assertSpanDurationCorrect(span, common.serverWait);
```

Comment:
For connect/express/restify I think the code in which spans are ended (`res.end` or equivalent) are run regardless of whether the request is aborted, suggesting that the server does the full amount of work without realizing the request was aborted.

Actually, reading more on this, I realize I should try this with multiple middleware.